### PR TITLE
Expose allowMissingPackageInstallation as a configuration property

### PR DESCRIPTION
**Pull request by kalymero**
Please note the `allowMissingPackageInstallation` parameter is not used internally and never passed down to the bakery, in any case so it has been just removed from the BakeRequest

### CODE REVIEW

File: orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfo.groovy [hunk 1]
```diff
@@ -44,11 +44,12 @@ class PackageInfo {
     potentialUrl ==~ /\b(https?|ssh):\/\/.*/
   }
 
-  public Map findTargetPackage() {
+  public Map findTargetPackage(boolean allowMissingPackageInstallation) {
     Map requestMap = [:]
     // copy the context since we may modify it in createAugmentedRequest
     requestMap.putAll(stage.execution.context)
     requestMap.putAll(stage.context)
+    requestMap['allowMissingPackageInstallation'] = allowMissingPackageInstallation
```
Comment (duftler) [inline orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfo.groovy]: Any reason not to pass this to `createAugmentedRequest()` as an argument, instead of adding it into the `requestMap`? Might read cleaner that way.

File: orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy [hunk 2]
```diff
@@ -110,12 +110,80 @@ class PackageInfoSpec extends Specification {
...
+    then:
+      targetPkg.packageVersion == packageVersion
+
+    where:
+      allowMissingPackageInstallation   ||  packageVersion
+      false                             ||  "1.1.1-h01.sha123"
+      true                              ||  "1.1.1-h01.sha123"
+  }
+
+  def "findTargetPackage: unmatched packages allowed when allowMissingPackageInstallation is true"() {
```
Comment (duftler) [inline orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy]: Any way this test and the next one can be combined, since the only real difference is the value of the new config property and the expected result of the `findTargetPackage()` call.
Seems like it would be a good use of `@Unroll`.

Comment (kalymero) [inline orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy]: I've tried but cannot find a way to combine them since one of them is throwing an exception.. any idea?

Comment (duftler) [inline orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy]: Maybe an `if (allowMissingPackageInstallation) { ... } else { ... }` within the `then:` block?
I _think_ that should work.

Comment (duftler) [inline orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy]: I've done something similar here: https://github.com/duftler/clouddriver/commit/1a7eab46fd99dacefe06b8c76b91815005f9124a#diff-d3ad83553d588351b1dd8293a124b548R143

Comment (kalymero) [inline orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy]: I've tried but it was complaining because the exception thing, let me try again

Comment (kalymero) [inline orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy]: I've found it's a common issue `Exception conditions are only allowed as top-level statements`... but there's a common pattern to solve it, applied

File: orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy [hunk 3]
```diff
@@ -110,12 +113,71 @@ class PackageInfoSpec extends Specification {
...
+      pipeline.context << [buildInfo: [artifacts: [[fileName: "api_1.1.1-h01.sha123_all.deb"]]]]
+      quipStage.execution = pipeline
+      quipStage.context = ['package': "another_package"]
+
+      PackageType packageType = PackageType.DEB
+      ObjectMapper objectMapper = new ObjectMapper()
+      PackageInfo packageInfo = new PackageInfo(quipStage, packageType.packageType, packageType.versionDelimiter, true, true, objectMapper )
+
+    expect:
+      Map targetPkg
```
Comment (duftler) [inline orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy]: Nice.

File: orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy [hunk 4]
```diff
@@ -110,12 +113,71 @@ class PackageInfoSpec extends Specification {
...
+    given:
+      Stage quipStage = new PipelineStage()
+      Pipeline pipeline = new Pipeline()
+      pipeline.context << [buildInfo: [artifacts: [[fileName: "api_1.1.1-h01.sha123_all.deb"]]]]
+      quipStage.execution = pipeline
+      quipStage.context = ['package': "another_package"]
+
+      PackageType packageType = PackageType.DEB
+      ObjectMapper objectMapper = new ObjectMapper()
+      PackageInfo packageInfo = new PackageInfo(quipStage, packageType.packageType, packageType.versionDelimiter, true, true, objectMapper )
```
Comment (duftler) [inline orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy]: Extraneous space just before closing parens.

File: orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy [hunk 5]
```diff
@@ -110,12 +113,71 @@ class PackageInfoSpec extends Specification {
...
+
+    expect:
+      Map targetPkg
+      try {
+        targetPkg = packageInfo.findTargetPackage(allowMissingPackageInstallation)
+        assert !expectedException
+
```
Comment (duftler) [inline orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy]: Extraneous blank line.

File: orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy [hunk 6]
```diff
@@ -110,12 +113,71 @@ class PackageInfoSpec extends Specification {
...
+      }
```
Comment (duftler) [inline orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy]: Formatting is off here:
`} catch (IllegalStateException ex) {`

**Comment by duftler** [comment]
LGTM after minor comments.
Please squash the commits as well.

**Comment by kalymero** [comment]
done

**Comment by duftler** [comment]
LGTM.

**Comment by duftler** [comment]
@tomaslin fyi
